src/mscorlib/src/System/Guid.cs

```diff
@@ -1322,6 +1360,123 @@ unsafe private static int HexsToCharsHexOutput(char* guidChars, int a, int b)
             return 9;
         }
 
+        // Fills a char* with the appropriate string. 
+        unsafe private void FormatGuid(char* guidChars, ref int offset, bool hex, bool dash)
```

Comment:
Realized offset is modified in FormatGuid but isn't used afterwards- changed the pass by ref. Thanks for the catch